packages/grafana-runtime/src/analytics/plugins/usePluginInteractionReporter.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  type DataSourcePluginContextType,
+  isDataSourcePluginContext,
+  type PluginContextType,
+  usePluginContext,
+} from '@grafana/data';
+
+import { reportInteraction } from '../utils';
+
+import { createDataSourcePluginEventProperties, createPluginEventProperties } from './eventProperties';
+
+export function usePluginInteractionReporter(): typeof reportInteraction {
+  const context = usePluginContext();
+
+  if (isDataSourcePluginContext(context)) {
+    return createDataSourceReporter(context);
+  }
+
+  return createPluginReporter(context);
+}
```

Comment:
Should we add some basic tests for this to see if it is returning the correct reporter?